polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -261,10 +261,18 @@ ValueCategory MLIRScanner::callHelper(
         } else
           Val = Arg.getValue(Builder);
 
-        if (isa<LLVM::LLVMPointerType>(Val.getType()) &&
-            isa<MemRefType>(ExpectedType))
-          Val = Builder.create<polygeist::Pointer2MemrefOp>(Loc, ExpectedType,
-                                                            Val);
+        if (auto PT = dyn_cast<LLVM::LLVMPointerType>(Val.getType())) {
+          if (auto MT = dyn_cast<MemRefType>(ExpectedType)) {
+            Val = Builder.create<polygeist::Pointer2MemrefOp>(
+                Loc,
+                MemRefType::get(MT.getShape(), MT.getElementType(), {},
```

Comment:
should we use `getLayout` like `MemRefType::get(MT.getShape(), MT.getElementType(), MT.getLayout(),`?

Reply:
Addressed. Nice one